homework/src/index.js

```diff
@@ -1,28 +1,31 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4) {
+          if (xhr.status < 400) {
+            resolve(xhr.response);
+          } else {
+            reject(new Error(xhr.statusText));
+          }
+        }
+      };
+      xhr.send();
+    });
```

Comment:
Why did you change this function from using `.onload` to `.onreadystatechange`? You are not handling network errors here, only server errors. In the original version network errors are handled through `.onerror`.

For instance, if I change the url to a non-existing domain name like this (note the extra `x` in `comx`:

```js
const HYF_REPOS_URL = 'https://api.github.comx/orgs/HackYourFuture/repos?per_page=100';
```

then I get this runtime error:

```
Cannot read property 'sort' of null
```